rag_experiment_accelerator/config/index_config.py

```diff
@@ -63,29 +82,18 @@ def index_name(self) -> str:
         return index_name
 
     @classmethod
-    def __get_index_value(cls, value: str) -> str:
-        return value.split("-")[1].strip()
-
-    @classmethod
-    def from_index_name(cls, index_name: str, config: Any) -> "IndexConfig":
+    def from_index_name(cls, index_name: str) -> "IndexConfig":
         """
         Creates IndexConfig from the index name.
         Reverse of index_name().
         """
-        values = index_name.split("_")
-        if len(values) != 11:
-            raise (f"Invalid index name [{index_name}]")
 
-        return IndexConfig(
-            index_name_prefix=values[0],
-            preprocess=bool(int(cls.__get_index_value(values[1]))),
-            chunk_size=int(cls.__get_index_value(values[2])),
-            overlap_size=int(cls.__get_index_value(values[3])),
-            ef_construction=int(cls.__get_index_value(values[4])),
-            ef_search=int(cls.__get_index_value(values[5])),
-            sampling_percentage=int(cls.__get_index_value(values[6])),
-            generate_title=bool(int(cls.__get_index_value(values[7]))),
-            generate_summary=bool(int(cls.__get_index_value(values[8]))),
-            override_content_with_summary=bool(int(cls.__get_index_value(values[9]))),
-            embedding_model=config._find_embedding_model_by_name(values[10].strip()),
-        )
+        key_values = [kv.split("-") for kv in index_name.split("_")]
```

Comment:
This is very dangerous function given the limitation of 128 characters from Azure for an index name.
Probably this mechanism shouldn't exist in the first place. I would suggest either to fix it or open a ticket with high priority that fixes this

Reply:
Created an issue #684